classes/Telemetry.php

```diff
@@ -0,0 +1,353 @@
+<?php
+/*******************************************************************************
+ * Copyright (c) 2019, Code Atlantic LLC
+ ******************************************************************************/
+
+if ( ! defined( 'ABSPATH' ) ) {
+	// Exit if accessed directly
+	exit;
+}
+
+/**
+ * Our telemetry class.
+ *
+ * Handles sending usage data back to our servers for those who have opted into our telemetry.
+ *
+ * @since 1.11.0
+ */
+class PUM_Telemetry {
+
+	public static function init() {
+		add_action( 'pum_daily_scheduled_events', array( __CLASS__, 'track_check' ) );
+		if ( is_admin() && current_user_can( 'manage_options' ) ) {
+			add_filter( 'pum_alert_list', array( __CLASS__, 'optin_alert' ) );
+			add_action( 'init', array( __CLASS__, 'optin_alert_check' ) );
+		}
+	}
+
+	/**
+	 * Prepares and sends data, if it is time to do so
+	 * @since 1.11.0
+	 */
+	public static function track_check() {
+		if ( self::is_time_to_send() ) {
+			$data = self::setup_data();
+			self::send_data( $data );
+			set_transient( 'pum_tracking_last_send', true, 6 * DAY_IN_SECONDS );
+		}
+	}
+
+	/**
+	 * Prepares telemetry data to be sent
+	 * @return array
+	 * @since 1.11.0
+	 */
+	public static function setup_data() {
+		global $wpdb;
+
+		// Retrieve current theme info
+		if ( get_bloginfo( 'version' ) < '3.4' ) {
```

Comment:
Probably could remove this. We don't support sites that old anyways haha.

Reply:
Not to create work, just a thought for future clean up.

Reply:
Yeah, I just left what was there from the old system. But, I agree that by now, we can probably remove that part. I'll go ahead and do that.

Reply:
Removed.